server/sonar-server/src/main/java/org/sonar/server/notification/DefaultNotificationManager.java

```diff
@@ -112,54 +106,42 @@ void logDeserializationIssue() {
   }
 
   public long count() {
-    return notificationQueueDao.count();
+    return dbClient.notificationQueueDao().count();
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
-  public Multimap<String, NotificationChannel> findSubscribedRecipientsForDispatcher(NotificationDispatcher dispatcher,
-    @Nullable String projectUuid) {
+  public Multimap<String, NotificationChannel> findSubscribedRecipientsForDispatcher(NotificationDispatcher dispatcher, String projectUuid) {
+    checkArgument(projectUuid != null, "ProjectUUID is mandatory");
     String dispatcherKey = dispatcher.getKey();
 
     SetMultimap<String, NotificationChannel> recipients = HashMultimap.create();
     for (NotificationChannel channel : notificationChannels) {
       String channelKey = channel.getKey();
 
       // Find users subscribed globally to the dispatcher (i.e. not on a specific project)
-      addUsersToRecipientListForChannel(propertiesDao.selectUsersForNotification(dispatcherKey, channelKey, null), recipients, channel);
+      // And users subscribed to the dispatcher specifically for the project
+      List<String> subscribedUsers = dbClient.propertiesDao().findUsersForNotification(dispatcherKey, channelKey, projectUuid);
 
-      if (projectUuid != null) {
-        // Find users subscribed to the dispatcher specifically for the project
-        addUsersToRecipientListForChannel(propertiesDao.selectUsersForNotification(dispatcherKey, channelKey, projectUuid), recipients, channel);
+      try (DbSession dbSession = dbClient.openSession(false)) {
```

Comment:
optimization: no need to open a DB connection nor to make a SQL call if `subscribedUsers` is empty